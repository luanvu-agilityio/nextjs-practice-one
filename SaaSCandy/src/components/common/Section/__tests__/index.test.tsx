import { render } from '@testing-library/react';
import { Section } from '../index';

describe('Section', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Section title='Test Title' subtitle='Test Subtitle'>
        <div>Content</div>
      </Section>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders without title and subtitle', () => {
    const { container } = render(
      <Section>
        <div>Content only</div>
      </Section>
    );
    expect(container.querySelector('section')).toBeInTheDocument();
  });

  it('applies centered class when centered prop is true', () => {
    const { container } = render(
      <Section title='Title' centered>
        <div>Content</div>
      </Section>
    );
    expect(container.querySelector('.text-center')).toBeInTheDocument();
  });
});
