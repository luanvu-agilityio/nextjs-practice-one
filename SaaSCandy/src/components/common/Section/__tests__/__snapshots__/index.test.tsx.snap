// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Section > matches snapshot 1`] = `
<div>
  <section
    class="py-12 sm:py-16 "
  >
    <div
      class="max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 "
    >
      <div
        class="mb-8 sm:mb-12 "
      >
        <h2
          aria-level="2"
          class="text-gray-900 font-bold mb-3 sm:mb-4 text-2xl sm:text-4xl lg:text-5xl"
          role="heading"
        >
          Test Title
        </h2>
        <p
          class="text-lg text-gray-600 leading-relaxed max-w-3xl mx-auto sm:text-lg"
          role="paragraph"
        >
          Test Subtitle
        </p>
      </div>
      <div>
        Content
      </div>
    </div>
  </section>
</div>
`;

exports[`Section matches snapshot 1`] = `
<div>
  <section
    class="py-12 sm:py-16 "
  >
    <div
      class="max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 "
    >
      <div
        class="mb-8 sm:mb-12 "
      >
        <h2
          aria-level="2"
          class="text-gray-900 font-bold mb-3 sm:mb-4 text-2xl sm:text-4xl lg:text-5xl"
          role="heading"
        >
          Test Title
        </h2>
        <p
          class="text-lg text-gray-600 leading-relaxed max-w-3xl mx-auto sm:text-lg"
          role="paragraph"
        >
          Test Subtitle
        </p>
      </div>
      <div>
        Content
      </div>
    </div>
  </section>
</div>
`;
